### PR TITLE
Den 3147 Reading and writing a serial number to the Orion board's EEPROM

### PR DESCRIPTION
**Pull request by CraigDensity**
Added functions to read and write a serial number to the FX2's I2C connected EEPROM on the Orion board.

### CODE REVIEW

File: cycfx2prog-0.47/cycfx2dev.h [hunk 1]
```diff
@@ -34,6 +34,8 @@ class CypressFX2Device
 		// Internally used: Program one line of an Intel HEX file. 
 		// The arguments path and line are just needed for error reporting. 
 		int _ProgramIHexLine(const char *buf,const char *path,int line);
+                // Internally used to calculate a serial number's checksum
+                unsigned short Computecrc16(const unsigned char* data_p, unsigned char length);
 	private:
```
Comment (bradfordbarr) [inline cycfx2prog-0.47/cycfx2dev.h]: Why are there two `private` labels?

Comment (CraigDensity) [reply cycfx2prog-0.47/cycfx2dev.h]: Good Point. I'll remove it

File: cycfx2prog-0.47/cycfx2dev.cc [hunk 2]
```diff
@@ -1069,6 +1069,150 @@ int CypressFX2Device::SerialNumberWrite(const unsigned char *ctl_buf, size_t ctl
         return(rv);
 }
 
+int CypressFX2Device::OrionSerialNumberRead(const unsigned char *ctl_buf, size_t ctl_buf_size)
```
Comment (ek9852) [inline cycfx2prog-0.47/cycfx2dev.cc]: const unsigned char => unsigned char ?

File: cycfx2prog-0.47/cycfx2dev.h [hunk 3]
```diff
@@ -124,6 +125,12 @@ class CypressFX2Device
                 // Write a serial number to the FX2 EEPROM
                 int SerialNumberWrite(const unsigned char *ctl_buf,
                         size_t ctl_buf_size);
+                // Read the serial number from the Orion board
+                int OrionSerialNumberRead(const unsigned char *ctl_buf,
```
Comment (ek9852) [inline cycfx2prog-0.47/cycfx2dev.h]: const unsigned char => unsigned char ?

File: cycfx2prog-0.47/cycfx2prog.cc [hunk 4]
```diff
@@ -712,7 +716,50 @@ int main(int argc,char **arg)
...
+                        }
+                        else
+                        {
+                             ++errors;
+                        }
+                }
+                else if(!strcmp(cmd,"setorionserial"))
+                {
+                    // if a serial number string has been provided and no more than 32 characters
+                    if(a[0] && *a[0] && (strlen(a[0]) < 33))
```
Comment (ek9852) [inline cycfx2prog-0.47/cycfx2prog.cc]: Not sure why it is 32 characters ? doesn't we write ascii 64 characters directly ?
Or we write binary 32 characters ?

Comment (CraigDensity) [reply cycfx2prog-0.47/cycfx2prog.cc]: Good Catch! Thanks Keith!

**Pull request merged**